src/NServiceBus.Core.Tests/ApprovalFiles/APIApprovals.ApproveNServiceBus.approved.txt

```diff
@@ -2312,38 +2341,78 @@ namespace NServiceBus.Transport
         "provided by transports. See the upgrade guide for more details. Will be removed " +
         "in version 9.0.0.", true)]
     public interface ICancelDeferredMessages
     {
         System.Threading.Tasks.Task CancelDeferredMessages(string messageKey, NServiceBus.Pipeline.IBehaviorContext context);
     }
+    [System.Obsolete("Queue creation is done by TransportDefinition.Initialize. Will be removed in vers" +
+        "ion 9.0.0.", true)]
     public interface ICreateQueues
     {
+        [System.Obsolete("Queue creation is done by TransportDefinition.Initialize. Will be removed in vers" +
+            "ion 9.0.0.", true)]
         System.Threading.Tasks.Task CreateQueueIfNecessary(NServiceBus.Transport.QueueBindings queueBindings, string identity);
     }
+    [System.Obsolete("The IDispatchMessages interface has been removed. See the upgrade guide for more " +
+        "details. Use `IMessageDispatcher` instead. Will be removed in version 9.0.0.", true)]
     public interface IDispatchMessages
     {
         System.Threading.Tasks.Task Dispatch(NServiceBus.Transport.TransportOperations outgoingMessages, NServiceBus.Transport.TransportTransaction transaction, NServiceBus.Extensibility.ContextBag context);
     }
+    [System.Obsolete("Use `ISubscriptionManager` instead. Will be removed in version 9.0.0.", true)]
     public interface IManageSubscriptions
     {
+        [System.Obsolete("Use `ISubscriptionManager.Subscribe` instead. Will be removed in version 9.0.0.", true)]
         System.Threading.Tasks.Task Subscribe(System.Type eventType, NServiceBus.Extensibility.ContextBag context);
+        [System.Obsolete("Use `ISubscriptionManager.Unsubscribe` instead. Will be removed in version 9.0.0." +
+            "", true)]
         System.Threading.Tasks.Task Unsubscribe(System.Type eventType, NServiceBus.Extensibility.ContextBag context);
     }
+    public interface IMessageDispatcher
+    {
+        System.Threading.Tasks.Task Dispatch(NServiceBus.Transport.TransportOperations outgoingMessages, NServiceBus.Transport.TransportTransaction transaction);
+    }
+    public interface IMessageReceiver
+    {
+        string Id { get; }
+        NServiceBus.Transport.ISubscriptionManager Subscriptions { get; }
+        System.Threading.Tasks.Task Initialize(NServiceBus.Transport.PushRuntimeSettings limitations, System.Func<NServiceBus.Transport.MessageContext, System.Threading.Tasks.Task> onMessage, System.Func<NServiceBus.Transport.ErrorContext, System.Threading.Tasks.Task<NServiceBus.Transport.ErrorHandleResult>> onError, System.Collections.Generic.IReadOnlyCollection<NServiceBus.Unicast.Messages.MessageMetadata> events);
+        System.Threading.Tasks.Task StartReceive();
+        System.Threading.Tasks.Task StopReceive();
+    }
     public interface IOutgoingTransportOperation
     {
-        System.Collections.Generic.List<NServiceBus.DeliveryConstraints.DeliveryConstraint> DeliveryConstraints { get; }
```

Comment:
should be obsoleted instead

Reply:
what would be our strategy to obsolete methods on interfaces? Technically, this should never be implemented by users but it might be because of testing reasons. Implementing an obsoleted method sounds off as well, so just remove it without obsolete?